zeebe/exporters/camunda-exporter/src/main/java/io/camunda/exporter/store/ExporterBatchWriter.java

```diff
@@ -0,0 +1,115 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Camunda License 1.0. You may not use this file
+ * except in compliance with the Camunda License 1.0.
+ */
+package io.camunda.exporter.store;
+
+import io.camunda.exporter.entities.ExporterEntity;
+import io.camunda.exporter.exceptions.PersistenceException;
+import io.camunda.exporter.handlers.ExportHandler;
+import io.camunda.zeebe.protocol.record.Record;
+import io.camunda.zeebe.protocol.record.RecordValue;
+import io.camunda.zeebe.protocol.record.ValueType;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+/** Caches exporter entities of different types and provide the method to flush them in a batch. */
+@SuppressWarnings({"rawtypes", "unchecked"})
+public class ExporterBatchWriter {
+  private final Map<ValueType, List<ExportHandler>> handlers = new HashMap<>();
+  private final Map<EntityIdAndEntityType, EntityAndHandler> cachedEntities = new HashMap<>();
```

Comment:
:question: Why should EntityType be part of the map key? Isn't `EntityId` unique?

Reply:
That's a good question, I also shared the same concern when looking at the draft PR. 
From what I could see in the implemented handlers there: I assume it should be unique, but it's done differently for each of the implemented handler and I wasn't quite sure of this assumption.

The way @sdorokhova  implemented this key was a `final Tuple<String, Class<?>> cacheKey =  new Tuple<>(id, handler.getEntityType());`. And I just replaced the `Tuple` with a record. I thought maybe she anticipates some cases where this id would be repeated for different record types but the entity type is what separates them.

E.g:  multiple records all has their id's as the `processInstanceKey` but one is a `ProcessInstanceRecord` and the other is a `SequenceFlow` for example. (this is not an accurate example, it's just for elaboration)

Reply:
Thanks. We can leave it as it is now and revisit later if we learn that this is not required. :+1: